tabsie/server.R

```diff
@@ -80,17 +80,48 @@ shinyServer(
     })
 ####### BUTTON PRESSES #####################
     observeEvent(input$clearTheme, {
-      if (!valAuth) return;#break processing of not authorized.
+      #if (!valAuth) return;#break processing of not authorized.
       updateTextInput(session, "titleField", value = "")
       updateTextInput(session, "xLab", value = "")
       updateTextInput(session, "yLab", value = "")
       updateSliderInput(session, "textSize", value = 15)
       updateSliderInput(session, "xLabRotation", value = 0)
       updateSliderInput(session, "xLabHeight", value = 0)
     })
+######## DIV BOX CONTROL for IDEAS and BUGS #################
+    observe({
+      #if (!valAuth) return();#break processing of not authorized.
+      shinyjs::onclick("toggleIdea",toggle(id="ideaDiv",anim=T))
+      shinyjs::onclick("toggleBug",toggle(id="bugDiv",anim=T))
+    })
+
+    observe({
+      #if (!valAuth) return();#break processing of not authorized.
+      needIdea <- need(input$ideaInput,F);
+      needBug <- need(input$bugInput,F);
+      if(length(c(needIdea,needBug))<=1) show('emailDiv',anim=T);
```

Comment:
Interesting R trivia fact... no matter how many nulls you put into a vector, it will be 0-length. It's as if you never bothered putting them in. Therefore, if you have a vector of length 2, that means neither needIdea nor needBug were nulls! Therefore neither of them passed the above need() tests. Otherwise, something got typed into at least one of the fields, and that means it's time to remind the user that we would like their email.